devel/cutter-r2/Portfile

```diff
@@ -0,0 +1,73 @@
+                    and FOSS reverse-engineering platform while keeping the user\
+                    experience at mind. Cutter is created by reverse engineers for\
+                    reverse engineers.
+
+homepage            https://cutter.re/
+
+checksums           rmd160  e2c88f2eb32c0ed6cee9e8cd3bc71ddf69363ed3 \
+                    sha256  4a65e5c2a02e0dd7d7359ede2f0ebb5c9d0e3411d6718f6aad57ba9affe53717 \
+                    size    1866181
+
+depends_lib-append  port:radare2
+qt5.depends_component \
+                    qtsvg
+                    
+compiler.cxx_standard 2011
+configure.args-append \
+                    src/Cutter.pro \
+                    INCLUDEPATH+=${prefix}/include/libr
+
+destroot {
+        copy ${worksrcpath}/Cutter.app ${destroot}${applications_dir}
+}
+
+variant python38 description {Enable Python support and bindings using Python 3.8} {
+        depends_lib-append  port:python38 \
+                            port:py38-pyside2
+
+        set python_framework \
+                            ${prefix}/Library/Frameworks/Python.framework
+
+        configure.args-append \
+                            CONFIG+=CUTTER_ENABLE_PYTHON \
+                            CONFIG+=CUTTER_ENABLE_PYTHON_BINDINGS \
+                            INCLUDEPATH+=${python_framework}/Versions/3.8/include/python3.8/ \
+                            LIBS+="-L${python_framework}/Versions/3.8/lib -lpython3.8" \
+                            QMAKE_RPATHDIR+=${python_framework}/Versions/3.8/lib/python3.8/site-packages/shiboken2/ \
+                            QMAKE_RPATHDIR+=${python_framework}/Versions/3.8/lib/python3.8/site-packages/PySide2/ \
+                            PYTHON_FRAMEWORK_DIR=${python_framework} \
+                            SHIBOKEN_EXECUTABLE=${python_framework}/Versions/3.8/bin/shiboken2 \
+                            SHIBOKEN_INCLUDEDIR=${python_framework}/Versions/3.8/lib/python3.8/site-packages/shiboken2_generator/include/ \
+                            SHIBOKEN_LIBRARY=${python_framework}/Versions/3.8/lib/python3.8/site-packages/shiboken2/libshiboken2.cpython-38-darwin.5.15.dylib \
+                            PYSIDE_INCLUDEDIR=${python_framework}/Versions/3.8/lib/python3.8/site-packages/PySide2/include/ \
+                            PYSIDE_LIBRARY=${python_framework}/Versions/3.8/lib/python3.8/site-packages/PySide2/libpyside2.cpython-38-darwin.5.15.dylib \
+                            PYSIDE_TYPESYSTEMS=${python_framework}/Versions/3.8/lib/python3.8/site-packages/PySide2/typesystems
+
+        pre-destroot {
+            move ${worksrcpath}/Cutter.app/Contents/MacOS/Cutter ${worksrcpath}/Cutter.app/Contents/MacOS/Cutter.bin
+            copy ${filespath}/Cutter ${worksrcpath}/Cutter.app/Contents/MacOS/
+            reinplace "s|@PYFRAMEWORK@|${prefix}/Library/Frameworks/Python.framework|g" ${worksrcpath}/Cutter.app/Contents/MacOS/Cutter
```

Comment:
That variable (and all variables set manually) is not accessible in `pre-destroot`, how to pass the variables in?

Reply:
Oh... possibly because you set the variable in a variant. Try setting the variable outside of the variant, in the main part of the portfile.

Reply:
That might work for `${python_framework}` but what about the variable for python version? Can't set that outside the variant.

Reply:
I haven't checked this. Are you sure using a variable that's set within the variant doesn't work? Since you would only be using the variable within the variant I thought that would be ok. I feel certain that we're doing this elsewhere.

Reply:
Yes it complains `no such variable`. I've found a workaround by setting the variable in global ns, will push later for you to check.